### PR TITLE
Adds docs for documentation portal.

### LINKED ISSUE #91: Generate reference docs according to a layout
### Description
We'd like to generate the reference documentation in this repo on every release. 

### Motivation
Have the documentation generated automatically.
Allow the documentation portal to pick up the new versions of the documentation on every release from this repo. 
Have reference docs in a layout that is consistent across the ecosystem.

### Requirements
- Documentation is generated in a *specified folder on every new release. 
- Documentation adheres to [specific layout](https://github.com/iotaledger/documentation/pull/94). Each method has its own .MD file generated. 

### Description
We'd like to generate the reference documentation in this repo on every release. 

### Motivation
Have the documentation generated automatically.
Allow the documentation portal to pick up the new versions of the documentation on every release from this repo. 
Have reference docs in a layout that is consistent across the ecosystem.

### Requirements
- Documentation is generated in a *specified folder on every new release. 
- Documentation adheres to [specific layout](https://github.com/iotaledger/documentation/pull/94). Each method has its own .MD file generated.

**Note:** We only have to keep the latest generated version. All previous versions will be persisted in the `iotaledger/documentation` repo.  

### *Folder structure
The docs should be generated in the following folders in this repo:
/docs/<project_name>/reference/<version>/<api_file>.MD

Later on, examples will be placed in:
/docs/<project_name>/examples/<api_file>.MD

### PR DESCRIPTION
**Pull request by luca-moser**
# Description

Adds a parser program which parses the documentation of all functions and generates markdown documentation according to the specified template. The parser will automatically pick up examples in given example dirs under each package and integrate them into the markdown file.

Fixes #91 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: docs/iota.go/reference/api_attach_to_tangle.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# API -> AttachToTangle()
+AttachToTangle performs the Proof-of-Work required to attach a transaction to the Tangle by calling the attachToTangle IRI API command. Returns a list of transaction trytes and overwrites the following fields:  Hash, Nonce, AttachmentTimestamp, AttachmentTimestampLowerBound, AttachmentTimestampUpperBound  If a Proof-of-Work function is supplied when composing the API, then that function is used instead of using the connected node.
```
Comment (jakubcech) [inline docs/iota.go/reference/api_attach_to_tangle.md]: Reckon some of these formatting things are caused by the way things are put together after parsing?:
"fields:  Hash,"
extra space
"AttachmentTimestampUpperBound  If a Proof-of-Work"
missing period after AttachmentTimestampUpperBound

Not a big deal, so if it's a hassle to fix then not to worry.

File: docs/iota.go/reference/api_broadcast_transactions.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# API -> BroadcastTransactions()
+BroadcastTransactions broadcasts a list of attached transaction trytes to the network. Tip-selection and Proof-of-Work must be done first by calling GetTransactionsToApprove and AttachToTangle or an equivalent attach method.  You may use this method to increase odds of effective transaction propagation.  Persist the transaction trytes in local storage before calling this command for first time, to ensure that reattachment is possible, until your bundle has been included.
+> **Important note:** This API is currently in Beta and is subject to change. Use of these APIs in production applications is not supported.
+
+
+## Input
+
+| Parameter       | Type | Required or Optional | Description |
+|:---------------|:--------|:--------| :--------|
+| trytes |  | true | The Trytes to broadcast.  |
```
Comment (jakubcech) [inline docs/iota.go/reference/api_broadcast_transactions.md]: Should this be a []Trytes type as well? In general, the Input part seems to be missing parameter types when it's a list of values. Like URIs in addNeighbors as well. Do we just want to do a [] if it's not typed to anything, like uris?

File: docs/iota.go/reference/api_check_consistency.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# API -> CheckConsistency()
+CheckConsistency checks if a transaction is consistent or a set of transactions are co-consistent.  Co-consistent transactions and the transactions that they approve (directly or indirectly), are not conflicting with each other and the rest of the ledger.  As long as a transaction is consistent, it might be accepted by the network. In case a transaction is inconsistent, it will not be accepted and a reattachment is required by calling ReplayBundle().
```
Comment (jakubcech) [inline docs/iota.go/reference/api_check_consistency.md]: 👍

File: docs/iota.go/reference/api_get_new_address.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# API -> GetNewAddress()
+GetNewAddress generates and returns a new address by calling FindTransactions and WereAddressesSpentFrom until the first unused address is detected. This stops working after a snapshot.  If the "total" parameter is supplied in the options, then this function simply generates the specified address range without doing any I/O.  It is suggested that the library user keeps track of used addresses and directly generates addresses from the stored information instead of relying on GetNewAddress.
```
Comment (jakubcech) [inline docs/iota.go/reference/api_get_new_address.md]: 👍

File: docs/iota.go/reference/api_get_new_address.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+> **Important note:** This API is currently in Beta and is subject to change. Use of these APIs in production applications is not supported.
+
+
+## Input
+
+| Parameter       | Type | Required or Optional | Description |
+|:---------------|:--------|:--------| :--------|
+| seed | Trytes | true | The seed from which to compute the addresses of.  |
```
Comment (jakubcech) [inline docs/iota.go/reference/api_get_new_address.md]: without the 'of'

File: docs/iota.go/reference/api_get_transactions_to_approve.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+# API -> GetTransactionsToApprove()
+GetTransactionsToApprove does the tip selection via the connected node.  Returns a pair of approved transactions which are chosen randomly after validating the transaction trytes, the signatures and cross-checking for conflicting transactions.  Tip selection is executed by a Random Walk (RW) starting at random point in the given depth, ending up to the pair of selected tips. For more information about tip selection please refer to the whitepaper (http://iotatoken.com/IOTA_Whitepaper.pdf).  The reference option allows to select tips in a way that the reference transaction is being approved too. This is useful for promoting transactions, for example with PromoteTransaction().
```
Comment (jakubcech) [inline docs/iota.go/reference/api_get_transactions_to_approve.md]: I would maybe link to our docs page, where the current implementation used in IRI is, but then we'll have to update for the new docs 🙈 
https://docs.iota.org/introduction/tangle/tip-selection
So maybe keep the whitepaper and when we're touching the part of the code from which this is generated, we can link to the new site if it's up by then.

File: docs/iota.go/reference/address_checksum.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+# Checksum()
+Checksum returns the checksum of the given address.
+> **Important note:** This API is currently in Beta and is subject to change. Use of these APIs in production applications is not supported.
+
+
+## Input
+
+| Parameter       | Type | Required or Optional | Description |
```
[the remark this answers is not in the document]
Comment (luca-moser) [reply docs/iota.go/reference/address_checksum.md]: Changed to Required/Optional

File: address/examples/address_examples_test.go [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	addr, err := address.GenerateAddress(seed, index, secLvl)
+	if err != nil {
+		log.Fatalf("unable to generate address: %s", err.Error())
+	}
+	fmt.Println(addr)
+	// output: GPB9PBNCJTPGFZ9CCAOPCZBFMBSMMFMARZAKBMJFMTSECEBRWMGLPTYZRAFKUFOGJQVWVUPPABLTTLCIA
+}
+
+// i req: seed, The seed used for address generation.
+// i req: start, The index from which to start to generate addresses from.
```
[the remark this answers is not in the document]
Comment (luca-moser) [reply address/examples/address_examples_test.go]: Fixed.

File: address/examples/address_examples_test.go [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+func ExampleValidChecksum() {
+	addr := "ZGPO9BSVZHJBLWHHRPOCKMRHLIEIOXQPOMGSDETZINIJGCDEP9QVJED9D9IUHNPPVDINQ9GOSLY9KWZGC"
+	checksum := "JHCYLIGUW"
+	if err := address.ValidChecksum(addr, checksum); err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+// i req: seed, The seed used for address generation.
+// i req: index, The index from which to generate the address from.
```
[the remark this answers is not in the document]
Comment (luca-moser) [reply address/examples/address_examples_test.go]: Fixed.

File: docs/iota.go/reference/api_attach_to_tangle.md (hunk 1 above)
Comment (luca-moser) [reply docs/iota.go/reference/api_attach_to_tangle.md]: Fixed. The comments on the function were wrong.

File: docs/iota.go/reference/api_broadcast_transactions.md (hunk 2 above)
Comment (luca-moser) [reply docs/iota.go/reference/api_broadcast_transactions.md]: Fixed. There was a problem where variadic parameters were not gathered correctly.

File: docs/iota.go/reference/api_get_transactions_to_approve.md (hunk 6 above)
Comment (luca-moser) [reply docs/iota.go/reference/api_get_transactions_to_approve.md]: I've basically copied this from iota.js. I agree that it is way too much detail.

**Comment by jakubcech** [comment]
Thanks, looks great. :)

**Pull request merged**